src/redux/createNetworkMiddleware.js

```diff
@@ -89,11 +98,12 @@ function createNetworkMiddleware({
   regexActionType = /FETCH.*REQUEST/,
   actionTypes = [],
   queueReleaseThrottle = 50,
+  shouldDequeueSelector = () => true,
 }: Arguments = {}) {
   return ({ getState }: MiddlewareAPI<State>) => (
     next: (action: any) => void,
   ) => (action: any) => {
-    const { isConnected, actionQueue } = getState().network;
+    const { isConnected, actionQueue, isQueuePaused } = getState().network;
```

Comment:
Tiny opt. Let's keep the state in a variable right away since we are calling `getState()` twice.

Reply:
So the reason for calling getState inside the `releaseQueue` loop is that this way the queue release function always has the latest `state` on every iteration. This way, when `CHANGE_QUEUE_SEMAPHORE` and  `CONNECTION_CHANGE` are dispatched the new state is reflected straight away inside `releaseQueue`, since at every iteration we call `getState()`. On the other hand, if we pass down `isQueuePaused` and `isConnected` from middleware function their values will only reflect the state when the `releaseQueue` was created, and will not update on further state change while the queue is being released. Does that sound right?